rllib/agents/sac/sac_model.py

```diff
@@ -111,17 +110,68 @@ def __init__(self,
         shift_and_log_scale_diag = tf.keras.layers.Concatenate(axis=-1)(
             [shift, log_scale_diag])
 
-        raw_action_distribution = tfp.layers.MultivariateNormalTriL(
-            self.action_dim)(shift_and_log_scale_diag)
+        batch_size = tf.keras.layers.Lambda(lambda x: tf.shape(input=x)[0])(
+            self.model_out)
+
+        base_distribution = tfp.distributions.MultivariateNormalDiag(
+            loc=tf.zeros(self.action_dim), scale_diag=tf.ones(self.action_dim))
+
+        latents = tf.keras.layers.Lambda(
+            lambda batch_size: base_distribution.sample(batch_size))(
+                batch_size)
+
+        self.shift_and_log_scale_diag = latents
+        self.latents_model = tf.keras.Model(self.model_out, latents)
+
+        def raw_actions_fn(inputs):
+            shift, log_scale_diag, latents = inputs
+            bijector = tfp.bijectors.Affine(
+                shift=shift, scale_diag=tf.exp(log_scale_diag))
```

Comment:
I seem to be getting an dimension incompatibility since this change when running SAC on any environment. I have tracked it down to here. Any idea @michaelzhiluo ?

Reply:
Not sure, what are your Tensorflow dependencies?

Reply:
Fixed here: https://github.com/ray-project/ray/pull/6642